packages/bitcore-client/bin/wallet-tx

```diff
@@ -18,24 +19,65 @@ program
   .option('--path [path]', 'optional - Custom wallet storage path')
   .parse(process.argv);
 
-const main = async () => {
-  const { name, path } = program;
-  let wallet;
-  try {
-    const { addresses, amount, utxos, change, fee } = program;
-    wallet = await Wallet.loadWallet({ name, path });
-    if (utxos) {
-      const password = await promptly.password('Wallet Password:');
-      wallet = await wallet.unlock(password);
+const loadWallet = async ({ name, path, password }) => {
+  const wallet = await Wallet.loadWallet({ name, path });
+  if (password) {
+    return await wallet.unlock(password);
+  }
+  return wallet;
+}
+
+const makeTx = async ({
+  name, path, addresses, amount, utxos, change, fee, password,
+}) => {
+  const wallet = await loadWallet({ name, path, password });
+  const tx = await wallet.newTx({ addresses, amount, utxos, change, fee });
+  console.log(tx.uncheckedSerialize());
+};
+
+const checkArgs = async (args) => {
+  for (const arg of [
+    'name', 'addresses', 'amount', 'utxos', 'change', 'fee'
+  ]) {
+    if (!args[arg]) {
+      throw new Erorr(`Did not specify --${arg}`);
     }
-    let params = { addresses, amount, utxos, change, fee };
-    params.addresses = addresses.includes('[') ? JSON.parse(addresses) : addresses;
-    params.utxos = params.utxos ? JSON.parse(params.utxos) : [];
-    const tx = await wallet.newTx(params);
-    console.log(tx.uncheckedSerialize());
-  } catch (e) {
-    console.error(e);
   }
 };
 
-main();
+(async () => {
+  const { name, path, addresses, amount, utxos, change, fee } = program;
+  const params = { name, path, addresses, amount, utxos, change, fee };
+  if (process.stdin.isTTY) {
+    checkArgs(params);
+  }
+
+  params.utxos = params.utxos ? JSON.parse(params.utxos) : [];
+  if (params.utxos) {
+    params.password = await promptly.password('Wallet Password:');
+  }
+
+  try {
+    params.addresses = JSON.parse(params.addresses);
+  }
+  catch (_) {}
+
+  if (process.stdin.isTTY) {
+    await makeTx(params);
+  }
+  else {
+    const rl = readline.createInterface({
+      input: process.stdin,
+      terminal: false
+    });
+    rl.on('line', async (line) => {
+      const parsed = JSON.parse(line);
+      if (parsed.utxos && !params.password) {
+        params.password = await promptly.password('Wallet Password:');
```

Comment:
I don't think this works :( I'm trying to do a mini version of this

```
#!/usr/bin/env node

const readline = require('readline');
const promptly = require('promptly');

(async function() {
  const rl = readline.createInterface({
    input: process.stdin,
    terminal: false
  });
  let password;
  rl.on('line', async (line) => {
    console.log(line);
    password = await promptly.password('Password: ');
  })
})();
```

echo "hi" | ./test 

Doesn't prompt for password